src/auth.py

```diff
@@ -347,29 +351,39 @@ def oauth_fb(request: Request) -> ResponseType:
         )
 
     token = user.get("token", "")
+    # Perform basic validation
+    if not token or len(token) > 1024 or not token.isalnum():
+        return jsonify({"status": "invalid", "msg": "Invalid Facebook token"}), 401
 
-    # Verify if this is a limited login (iOS only)
+    # Check whether this is a limited login (used by iOS clients only)
     is_limited_login = rq.get_bool("isLimitedLogin", False)
     if is_limited_login:
         # Validate Limited Login token
         try:
-            # Get the public key from Facebook's JWKS endpoint
+            # Get the (cached) public key from Facebook's JWKS endpoint
             public_key = get_facebook_public_key()
+            if not public_key:
+                return (
+                    jsonify(
+                        {"status": "invalid", "msg": "Missing Facebook public key",}
+                    ), 
+                    401,
+                )
             decoded_token = jwt.decode(
                 token,
                 public_key,
-                algorithms=['RS256'],
+                algorithms=["RS256"],
                 audience=facebook_app_id,
                 options={"verify_exp": True}
             )
-            if decoded_token.get('nonce') != FACEBOOK_NONCE:
+            if decoded_token.get("nonce") != FACEBOOK_NONCE:
                 return (
                     jsonify(
                         {"status": "invalid", "msg": "Invalid Facebook nonce token",}
                     ), 
                     401,
                 )
-            account = decoded_token.get('sub', '')
+            account = decoded_token.get("sub", "")
```

Comment:
In the other case (non-limited login), the account id is compared with the original string from the login request, and if different, the login is aborted. Does the same logic apply here, in the limited login case? @kadyvillicana